test/metabase/api/preview_embed_test.clj

```diff
@@ -141,6 +141,20 @@
              (mt/user-http-request :crowberto :get 202 (str (card-query-url card {:_embedding_params {:venue_id "enabled"}})
                                                             "?venue_id=200")))))))))
 
+(deftest query-max-results-constraint-test
+  (testing "GET /api/preview_embed/card/:token/query"
+    (testing "Only 2000 results returned when there are many more"
+      (embed-test/with-embedding-enabled-and-new-secret-key
+        (let [;; this is the same sample question as setup in these steps to reproduce:
+              ;; https://github.com/metabase/metabase/issues/20938
+              sample-db-orders-question {:database 1
+                                         :query {:source-table 2}
+                                         :type "query"}]
```

Comment:
You can't assume that this DB exists yet or that it's ID is going to be `1`... what ID it gets depends on the order tests are ran. As proof, try

```sh
clojure -X:dev:test :only metabase.api.preview-embed-test/query-max-results-constraint-test
```

I'm not sure which Table you expect to get for ID `2`, but you want to do something like

`(mt/dataset sample-dataset ...)` and `(mt/mbql-query <whatever-table-name>)`. Grep the codebase for `mt/dataset` and you should find tons of examples

Reply:
Of course this only works if the table in question has > 2000 rows